GliaWidgets/Sources/CoreSDKClient/Logger/Logger.Interface.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+extension CoreSdkClient {
+    struct Logger {
+        var oneTimeClosure: () -> Logger
+        var prefixedClosure: (String) -> Logger
+        var localLoggerClosure: () -> Logger?
+        var remoteLoggerClosure: () -> Logger?
+        var errorClosure: (_ object: @autoclosure () -> Any, _ file: String, _ function: String, _ line: Int) -> Void
+        var warningClosure: (_ object: @autoclosure () -> Any, _ file: String, _ function: String, _ line: Int) -> Void
+        var infoClosure: (_ object: @autoclosure () -> Any, _ file: String, _ function: String, _ line: Int) -> Void
+        var debugClosure: (_ object: @autoclosure () -> Any, _ file: String, _ function: String, _ line: Int) -> Void
+        var configureLocalLogLevelClosure: (LogLevel) throws -> Void
+        var configureRemoteLogLevelClosure: (LogLevel) throws -> Void
+        var reportDeprecatedMethodClosure: (_ context: String, _ file: String, _ function: String, _ line: Int) -> Void
+
+        var oneTime: Logger {
+            oneTimeClosure()
+        }
+
+        func prefixed(_ prefix: String) -> Logger {
+            prefixedClosure(prefix)
+        }
+
+        func prefixed<TypeAsPrefix>(_ prefix: TypeAsPrefix.Type) -> Logger {
+            prefixed("\(prefix)")
+        }
+
+        var localLogger: Logger? {
+            localLoggerClosure()
+        }
+        var remoteLogger: Logger? {
+            remoteLoggerClosure()
+        }
+
+        func error(_ object: @autoclosure () -> Any, file: String = #fileID, function: String = #function, line: Int = #line) {
+            errorClosure(object(), file, function, line)
```

Comment:
the same question as above about passing closure or value. would we able to optimize this?